### PR TITLE
Add option: Add user to training; Some minor changes

### PR DESCRIPTION
**Pull request by jacentyg**

### CODE REVIEW

File: app/controllers/participations_controller.rb [hunk 1]
```diff
@@ -14,9 +25,26 @@ def destroy
...
+  def authorize_operator_create
+    training = Training.find(params[:training_id])
+    redirect_to root_url if current_user != training.operator &&
+                            !current_user.admin?
+  end
+
+  def authorize_operator_destroy
     participation = Participation.find(params[:id])
-    redirect_to root_url if current_user != participation.training.operator &&
+    training = Training.find(participation.training_id)
```
Comment (grzegorzhauska) [inline app/controllers/participations_controller.rb]: `participation.training`

File: app/controllers/participations_controller.rb [hunk 2]
```diff
@@ -14,9 +25,26 @@ def destroy
...
   private
 
-  def authorize_operator
+  def participation_params
+    params.require(:participation).permit(:user_id, :training_id)
+  end
+
+  def create_participation_params
+    participation_params.merge(training_id: params[:training_id])
+  end
```
Comment (grzegorzhauska) [inline app/controllers/participations_controller.rb]: `training_id` juz jest w metodzie `participation_params`

File: app/controllers/participations_controller.rb [hunk 3]
```diff
@@ -1,7 +1,18 @@
...
-  before_action :authorize_operator
+  before_action :authorize_operator_create, only: [:create]
+  before_action :authorize_operator_destroy, only: [:destroy]
+
+  def create
+    participation = Participation.new(create_participation_params)
+    user = User.find(params[:participation][:user_id])
+    training = Training.find(params[:training_id])
+    if participation.save
+      flash[:notice] = "#{user.full_name} has been added to training."
```
Comment (grzegorzhauska) [inline app/controllers/participations_controller.rb]: `participation.user`, da rade `participation.training` ?

File: app/controllers/participations_controller.rb (hunk 1 above)
Comment (jacentyg) [inline app/controllers/participations_controller.rb]: :+1:

File: app/controllers/participations_controller.rb [hunk 4]
```diff
@@ -1,7 +1,17 @@
...
+  before_action :authorize_operator_destroy, only: [:destroy]
+
+  def create
+    participation = Participation.new(participation_params)
+    if participation.save
+      flash[:notice] = "#{participation.user.full_name} has been added to
+        training."
+    end
+    redirect_to participation.training
+  end
```
Comment (jacentyg) [inline app/controllers/participations_controller.rb]: > grzegorzhauska 21 hours ago
> participation.user, da rade participation.training ?

:+1:

File: app/controllers/participations_controller.rb [hunk 5]
```diff
@@ -14,7 +24,19 @@ def destroy
 
   private
 
-  def authorize_operator
+  def participation_params
+    params.require(:participation).permit(:user_id, :training_id)
+  end
```
Comment (jacentyg) [inline app/controllers/participations_controller.rb]: > +  def create_participation_params
> +    participation_params.merge(training_id: params[:training_id])
> +  end
>       @grzegorzhauska
> grzegorzhauska 21 hours ago
> training_id juz jest w metodzie participation_params

:+1:

File: app/controllers/participations_controller.rb [hunk 6]
```diff
@@ -14,9 +23,21 @@ def destroy
...
+    params_id = params[:id]
+    training =
+      if params_id.nil?
+        Training.find(params[:participation][:training_id])
+      else
+        Participation.find(params_id).training
+      end
+    redirect_to root_url unless
+      helpers.current_user_admin_or_operator_for(training)
   end
```
Comment (grzegorzhauska) [inline app/controllers/participations_controller.rb]: nadal jest hujowo.

```
training =  if params[:id]
                   Participation.find(params[:id]).training
                 else
                   Training.find(params[:participation][:training_id])
                 end
```

albo sprobuj tak jak mowiles rozbic authorize operatora i admina na osobne metody albo dodaj admina do nazwy metody

**Pull request merged**